cpp/src/arrow/types/construct.h

```diff
@@ -20,19 +20,24 @@
 
 #include <cstdint>
 #include <memory>
-
+#include <vector>
 namespace arrow {
 
 class Array;
 class ArrayBuilder;
 class Buffer;
 struct DataType;
+struct Field;
 class MemoryPool;
 class Status;
 
 Status MakeBuilder(MemoryPool* pool, const std::shared_ptr<DataType>& type,
     std::shared_ptr<ArrayBuilder>* out);
 
+Status MakeStructBuilder(MemoryPool* pool, const std::shared_ptr<DataType>& type,
```

Comment:
Why do we need a new API? The fields are in the data type (need to expose a `children()` public API in DataType).